docs/resources/domain_record.md

```diff
@@ -0,0 +1,240 @@
+  dns_zone = "domain.tld"
+  name     = "images"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+
+  geo_ip {
+    matches {
+      continents = ["EU"]
+      countries  = ["FR"]
+      data       = "1.2.3.5"
+    }
+    
+    matches {
+      continents = ["NA"]
+      data       = "4.3.2.1"
+    }
+  }
+}
+
+resource "scaleway_domain_record" "http_service" {
+  dns_zone = "domain.tld"
+  name     = "app"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+
+  http_service {
+    ips          = ["1.2.3.5", "1.2.3.6"]
+    must_contain = "up"
+    url          = "http://mywebsite.com/health"
+    user_agent   = "scw_service_up"
+    strategy     = "hashed"
+  }
+}
+
+resource "scaleway_domain_record" "view" {
+  dns_zone = "domain.tld"
+  name     = "db"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+
+  view {
+    subnet = "100.0.0.0/16"
+    data   = "1.2.3.5"
+  }
+
+  view {
+    subnet = "100.1.0.0/16"
+    data   = "1.2.3.6"
+  }
+}
+
+resource "scaleway_domain_record" "weighted" {
+  dns_zone = "domain.tld"
+  name     = "web"
+  type     = "A"
+  data     = "1.2.3.4"
+  ttl      = 3600
+
+  weighted {
+    ip     = "1.2.3.5"
+    weight = 1
+  }
+
+  weighted {
+    ip     = "1.2.3.6"
+    weight = 2
+  }
+}
+```
+
+### Create an instance and add records with the new instance IP
+
+```hcl
+variable "project_id" {
+  type        = string
+  description = "Your project ID."
+}
+
+variable "dns_zone" {
+  type        = string
+  description = "The DNS Zone used for testing records."
+}
+
+resource "scaleway_instance_ip" "public_ip" {
+  project_id = var.project_id
+}
+
+resource "scaleway_instance_server" "web" {
+  project_id = var.project_id
+  type       = "DEV1-S"
+  image      = "ubuntu_focal"
+  tags       = ["front", "web"]
+  ip_id      = scaleway_instance_ip.public_ip.id
+
+  root_volume {
+    size_in_gb = 20
+  }
+}
+
+resource "scaleway_domain_record" "web_A" {
+  dns_zone = var.dns_zone
+  name     = "web"
+  type     = "A"
+  data     = scaleway_instance_server.web.public_ip
+  ttl      = 3600
+}
+
+resource "scaleway_domain_record" "web_cname" {
+  dns_zone = var.dns_zone
+  name     = "www"
+  type     = "CNAME"
+  data     = "web.${var.dns_zone}."
+  ttl      = 3600
+}
+
+resource "scaleway_domain_record" "web_alias" {
+  dns_zone = var.dns_zone
+  name     = ""
+  type     = "ALIAS"
+  data     = "web.${var.dns_zone}."
+  ttl      = 3600
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+- `dns_zone` - (Required) The DNS Zone of the domain. If the DNS zone doesn't exist, it will be automatically created.
+
+- `keep_empty_zone` - (Optional, default: `false`) When destroying a resource, if only NS records remain and this is set to `false`, the zone will be deleted. Please note, each zone not deleted will [cost you money](https://www.scaleway.com/en/dns/)
+
+- `name` - (Required) The name of the record (can be an empty string for a root record).
```

Comment:
If `name` can be empty should optional instead

Reply:
We are using the same convention as the API, empty is a value for us. No value != empty value imo